### PR TITLE
Support deserializing NaN, Infinity and -Infinity to corresponding Go values

### PR DESCRIPTION
**Pull request by vrana**

### CODE REVIEW

**Comment by vrana** [comment]
@cybrcodr

File: jsonpb/jsonpb.go [hunk 1]
```diff
@@ -971,6 +971,20 @@ func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMe
 		inputValue = inputValue[1 : len(inputValue)-1]
 	}
 
+	// Non-finite numbers can be encoded as strings.
+	isFloat := targetType.Kind() == reflect.Float32 || targetType.Kind() == reflect.Float64
+	if isFloat {
+		nonFinite := map[string]float64{
```
Comment (cybrcodr) [inline jsonpb/jsonpb.go]: Minor.  Let's define this at package level.

**Comment by vrana** [comment]
Done.

File: jsonpb/jsonpb.go (hunk 1 above)
Comment (vrana) [inline jsonpb/jsonpb.go]: Done.

**Pull request merged**